src/modules/rules/rules.service.ts

```
@@ -0,0 +1,59 @@
+    private readonly ruleRepository: Repository<Rule>,
+  ) {}
+
+  private validateThresholds(dto: CreateRuleDto | UpdateRuleDto) {
+    if (dto.similarityThreshold !== undefined && dto.deviationAllowedPct !== undefined) {
+      throw new BadRequestException('similarityThreshold and deviationAllowedPct cannot both be set');
+    }
+    if (dto.pattern) {
+      try {
+        // eslint-disable-next-line no-new
+        new RegExp(_.escapeRegExp(dto.pattern));
+      } catch (e) {
```

ESLint

[error] 'e' is defined but never used.

(@typescript-eslint/no-unused-vars)